### ISSUE TITLE
yolo v2 input size

### ISSUE THREAD

**Issue by 89douner** [AlexeyAB/darknet#879]
In yolo v2 paper, they described that yolo v2 receive 448 input image to train for detection. But, actually, when I tried to train yolo v2 through my customized labeling data, I used various input image size like 403x506, 664x305 and so on...

Why is the paper's input size different from the darknet model's input size?
Do darknet model resize the customized input image size to 448x448 ??

**Comment by ahsan856jalal** [comment]
yes , it resizes the batch before passing it to optimizer. If you set random=1 in the cfg file, then after every 10 iterations , it will resize the images for the next 10 iterations from the explained sizes in the yolov2 paper

**Comment by AlexeyAB** [comment]
> But, actually, when I tried to train yolo v2 through my customized labeling data, I used various input image size like 403x506, 664x305 and so on...

Any image will be resized to the network size (width= height= in your cfg file).
Default network size of the Yolo v2 is 416x416 (of Yolo v1 is 448x448).

----

* **classifier Darknet19_448** is trained by using **448x448** network size

* **detector Yolo v2** is trained by using **416x416** network size (by using classifier Darknet19_448 pre-trained weights-file)


https://arxiv.org/pdf/1612.08242v1.pdf

> For YOLOv2 we first fine tune the classification network
> at the full 448 × 448 resolution for 10 epochs on ImageNet

> We also
> shrink the network to operate on 416 input images instead
> of 448×448.
